src/components/Input/index.tsx

```diff
@@ -97,30 +157,84 @@ export const Input = React.forwardRef<HTMLInputElement, InputProps>(
       block,
       defaultValue,
       id,
+      onClickCopy = null,
+      shouldShowHideShowButton,
       ...props
     },
     ref
   ) => {
+    const [type, setType] = useState(props.type || 'text')
+    const [wasClicked, setWasClicked] = useState(false)
     const testId = props['data-testid']
     const shouldShowErrorState =
       !(error === false || error === '' || error === undefined) || isUnavailable
     const shouldDisplayInfo = !shouldShowErrorState && info
     const errorMessage = typeof error === 'string' ? error : null
 
+    useEffect(() => {
+      if (wasClicked) {
+        setTimeout(() => {
+          setWasClicked(false)
+        }, 800)
+      }
+    }, [wasClicked])
+
     return (
       <InputContainer $block={block}>
         <Label htmlFor={id} $small={hasSmallLabel}>
           {label}
-          <AsertoInput
-            ref={ref}
-            isValid={isValid && !shouldShowErrorState}
-            isInvalid={shouldShowErrorState}
-            $isUnavailable={isUnavailable}
-            onChange={onChange}
-            style={style}
-            defaultValue={defaultValue}
-            {...props}
-          />
+          <InputButtonBlock>
+            <AsertoInput
+              ref={ref}
+              isValid={isValid && !shouldShowErrorState}
+              isInvalid={shouldShowErrorState}
+              $isUnavailable={isUnavailable}
+              onChange={onChange}
+              style={style}
+              defaultValue={defaultValue}
+              {...props}
+              type={type}
+            />
+            {!(error || isUnavailable || isValid) && (
+              <ButtonsContainer>
+                {shouldShowHideShowButton && (
+                  <>
+                    {type === 'password' ? (
+                      <HideShowButton
+                        variant="secondary-borderless"
+                        onClick={() => setType('text')}
+                        {...mapTestIdToProps(`${testId}-show-btn`)}
+                      >
+                        <img alt="show" src={show} />
+                      </HideShowButton>
+                    ) : (
+                      <HideShowButton
+                        onClick={() => setType('password')}
+                        variant="secondary-borderless"
+                        {...mapTestIdToProps(`${testId}-hide-btn`)}
```

Comment:
I'm not a fan of maintaining these testid fields. Consumers of the library shouldn't be testing against implementation details.

Reply:
ok I can remove all references to `testId` as a whole in this file

Reply:
we should keep the already existing ones to ensure that we don't break anything depending on them. we should remove those dependencies and then we can remove the remaining ones, which we should version as a breaking change.

Reply:
so just in copy and hide/show?

Reply:
yeah